api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -752,12 +781,24 @@ private Optional<String> buildConceptIdSqlInClause(
             .collect(Collectors.toList());
     final List<Long> dbConceptSetIds =
         conceptSets.stream().map(DbConceptSet::getConceptSetId).collect(Collectors.toList());
-    List<DbConceptSetConceptId> dbConceptSetConceptIds = new ArrayList<>();
+    Set<DbConceptSetConceptId> dbConceptSetConceptIds = new HashSet<>();
+    List<Long> surveyConceptIds = new ArrayList<>();
     List<Long> dbCriteriaAnswerIds = new ArrayList<>();
     if (domain.equals(Domain.SURVEY)) {
-      if (!prePackagedSurveyConceptSet(dbConceptSets)) {
-        dbConceptSetConceptIds = findDomainConceptIds(domain, dbConceptSetIds);
+      if (prePackagedAllSurveyConceptSet(dbConceptSets)) {
+        return Optional.empty();
+      }
+      if (userSurveyConceptSet(dbConceptSets)) {
+        dbConceptSetConceptIds.addAll(findDomainConceptIds(domain, dbConceptSetIds));
       }
+      // handle prepackaged PFHH
+      if (prePackagedPfhhSurveyConceptSet(dbConceptSets)) {
```

Comment:
Shouldn't this be behind the feature flag?

if (workbenchConfigProvider.get().featureFlags.enableDataExplorer)